python/lsst/ts/ess/common/device/mock_boltek_formatter.py

```diff
@@ -0,0 +1,124 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "MockEFM100CFormatter",
+    "MockLD250NoiseFormatter",
+    "MockLD250StatusFormatter",
+    "MockLD250StrikeFormatter",
+]
+
+import random
+
+from ..constants import NOISE_PREFIX, STATUS_PREFIX, STRIKE_PREFIX
+from .mock_formatter import (
+    MockAzimuthConfig,
+    MockDistanceConfig,
+    MockElectricFieldStrengthConfig,
+    MockFormatter,
+    MockStrikeRateConfig,
+)
+
+
+def random_checksum() -> str:
+    """Return a 2 digit random hex string that serves as a checksum."""
+    return f"{random.randint(0, 255):02X}"
+
+
+class MockEFM100CFormatter(MockFormatter):
+    def __init__(self) -> None:
+        self.in_error_state = False
+
+    def format_output(
+        self,
+        num_channels: int = 0,
+        disconnected_channel: int = 0,
+        missed_channels: int = 0,
+    ) -> list[str]:
+        efs = random.uniform(
+            MockElectricFieldStrengthConfig.min, MockElectricFieldStrengthConfig.max
+        )
+        fault = 1 if self.in_error_state else 0
+        return [f"${efs:+06.2f}", f"{fault}*{random_checksum()}"]
+
+
+class MockLD250NoiseFormatter:
+    """Format a string representing noise telemetry emitted by a Boltek LD-250
+    sensor.
+
+    This class doesn't need to be a subclass of `MockFormatter` since it
+    doesn't get initialized in the formatter registry of the `MockDevice`
+    class.
```

Comment:
A minor point, but I am uneasy about this note.

I assume you are using this naming convention (which implies the subclass relationship) because it meets the API requirements -- the format_output method? In that case...perhaps just inherit, even though it is not necessary? Then you could modify the note to say that the inheritance is not strictly necessary, but was done for API uniformity.

If "acting like a MockFormatter" is not actually helpful to your code, then I'd suggest making this class (and the other one like it below) into bare functions whose names did not imply a subclass of MockFormatter.

Reply:
Good point. I made them subclasses of MockFormatter and didn't add them to the registry.